### PR TITLE
[4.0] Namespace

### PR DESCRIPTION
**Pull request by brianteeman**
I really hope this is the last time I have to do this namespsacing

over to you @quy to fix my errors

### CODE REVIEW

File: administrator/components/com_content/Field/VoteradioField.php [hunk 1]
```diff
@@ -12,8 +12,9 @@
 defined('_JEXEC') or die;
 
 use Joomla\CMS\Plugin\PluginHelper;
+use Joomla\CMS\Form\FormHelper;
```
Comment (Quy) [inline administrator/components/com_content/Field/VoteradioField.php:14]: Alpha order.

File: administrator/components/com_modules/Controller/ModuleController.php [hunk 2]
```diff
@@ -234,7 +234,7 @@ public function save($key = null, $urlVar = null)
 			$this->input->post->set('jform', $data);
 
 			// Add path of forms directory
-			\JForm::addFormPath(JPATH_ADMINISTRATOR . '/components/com_modules/models/forms');
+			Form::addFormPath(JPATH_ADMINISTRATOR . '/components/com_modules/models/forms');
```
Comment (Quy) [inline administrator/components/com_modules/Controller/ModuleController.php:238]: Add `use Joomla\CMS\Form\Form;`

File: administrator/modules/mod_privacy_dashboard/mod_privacy_dashboard.php [hunk 3]
```diff
@@ -9,25 +9,30 @@
 
 defined('_JEXEC') or die;
 
+use Joomla\CMS\Factory;
+use Joomla\CMS\Helper\ModuleHelper;
+use Joomla\CMS\HTML\HTMLHelper;
+use Joomla\CMS\Router\Route;
```
Comment (Quy) [inline administrator/modules/mod_privacy_dashboard/mod_privacy_dashboard.php]:
```suggestion
```

File: administrator/templates/atum/login.php [hunk 4]
```diff
@@ -86,7 +86,7 @@
 								<h2 class="m-0 text-primary"><?php echo $sitename; ?></h2>
 							<?php endif; ?>
 							<?php if ($loginLogo) : ?>
-								<img src="<?php echo JURI::root() . '/' . $loginLogo; ?>" class="img-fluid my-2" alt="">
+								<img src="<?php echo URI::root() . '/' . $loginLogo; ?>" class="img-fluid my-2" alt="">
```
Comment (Quy) [inline administrator/templates/atum/login.php]:
```suggestion
								<img src="<?php echo Uri::root() . '/' . $loginLogo; ?>" class="img-fluid my-2" alt="">
```

File: components/com_finder/helpers/route.php [hunk 5]
```diff
@@ -9,6 +9,10 @@
 
 defined('_JEXEC') or die;
 
+use Joomla\CMS\Factory;
+use Joomla\CMS\Component\ComponentHelper;
```
Comment (Quy) [inline components/com_finder/helpers/route.php:12]: Alpha order

File: components/com_finder/tmpl/search/default_result.php [hunk 6]
```diff
@@ -10,10 +10,12 @@
 defined('_JEXEC') or die;
 
 use Joomla\CMS\HTML\HTMLHelper;
+use Joomla\CMS\Factory;
```
Comment (Quy) [inline components/com_finder/tmpl/search/default_result.php:12]: Alpha order

File: libraries/cms/html/uitab.php [hunk 7]
```diff
@@ -39,7 +41,7 @@ public static function startTabSet($selector = 'myTab', $params = array())
 		if (!isset(static::$loaded[__METHOD__][$sig]))
 		{
 			// Include the custom element
-			JHtml::_('webcomponent', 'vendor/joomla-custom-elements/joomla-tab.min.js', ['version' => 'auto', 'relative' => true]);
+			JHHTMLHelpertml::_('webcomponent', 'vendor/joomla-custom-elements/joomla-tab.min.js', ['version' => 'auto', 'relative' => true]);
```
Comment (Quy) [inline libraries/cms/html/uitab.php]:
```suggestion
			HTMLHelper::_('webcomponent', 'vendor/joomla-custom-elements/joomla-tab.min.js', ['version' => 'auto', 'relative' => true]);
```

File: plugins/system/debug/DataCollector/InfoCollector.php [hunk 8]
```diff
@@ -14,6 +14,7 @@
 use Joomla\CMS\Application\SiteApplication;
 use Joomla\CMS\Factory;
 use Joomla\CMS\User\User;
+use Joomla\CMS\Uri\Uri;
```
Comment (Quy) [inline plugins/system/debug/DataCollector/InfoCollector.php:16]: Alpha order

File: components/com_finder/tmpl/search/default_result.php (hunk 6 above)
Comment (Quy) [reply components/com_finder/tmpl/search/default_result.php:12]: Ping

File: installation/src/Model/DatabaseModel.php [hunk 9]
```diff
@@ -11,6 +11,7 @@
 
 defined('_JEXEC') or die;
 
+use Joomla\Database\DatabaseDriver;
```
Comment (Quy) [inline installation/src/Model/DatabaseModel.php]: Alpha order

File: libraries/src/Router/ApiRouter.php [hunk 10]
```diff
@@ -14,6 +14,7 @@
 use Joomla\CMS\Router\Exception\RouteNotFoundException;
 use Joomla\Router\Router;
 use Joomla\Router\Route;
+use Joomla\CMS\Uri\Uri;
```
Comment (Quy) [inline libraries/src/Router/ApiRouter.php]: Alpha order

File: libraries/src/Application/ApiApplication.php [hunk 11]
```diff
@@ -12,12 +12,14 @@
 
 use Joomla\Application\Web\WebClient;
 use Joomla\CMS\Access\Exception\AuthenticationFailed;
+use Joomla\CMS\Factory;
 use Joomla\CMS\Router\Exception\RouteNotFoundException;
 use Joomla\CMS\Plugin\PluginHelper;
-use Joomla\Input\Json as JInputJson;
 use Joomla\CMS\Component\ComponentHelper;
```
Comment (Quy) [inline libraries/src/Application/ApiApplication.php]: Alpha order

File: libraries/src/Application/ApiApplication.php [hunk 12]
```diff
@@ -12,12 +12,14 @@
...
 use Joomla\CMS\Router\ApiRouter;
+use Joomla\CMS\Uri\Uri;
 use Joomla\DI\Container;
+use Joomla\Input\Json as JInputJson;
```
Comment (Quy) [inline libraries/src/Application/ApiApplication.php]: Rename `JInputJson`

File: libraries/src/Application/ApiApplication.php (hunk 11 above)
Comment (Quy) [reply libraries/src/Application/ApiApplication.php]: Sort lines 16-19  then we should be good to go.

File: libraries/src/Application/ApiApplication.php [hunk 13]
```diff
@@ -12,12 +12,14 @@
 
 use Joomla\Application\Web\WebClient;
 use Joomla\CMS\Access\Exception\AuthenticationFailed;
-use Joomla\CMS\Router\Exception\RouteNotFoundException;
-use Joomla\CMS\Plugin\PluginHelper;
-use Joomla\Input\Json as JInputJson;
+use Joomla\CMS\Factory;
```
Comment (Quy) [inline libraries/src/Application/ApiApplication.php]:
```suggestion
```

File: libraries/src/Application/ApiApplication.php [hunk 14]
```diff
@@ -12,12 +12,14 @@
...
 use Joomla\CMS\Component\ComponentHelper;
```
Comment (Quy) [inline libraries/src/Application/ApiApplication.php:15]:
```suggestion
use Joomla\CMS\Component\ComponentHelper;
use Joomla\CMS\Factory;
```

File: administrator/components/com_menus/Model/MenuModel.php [hunk 15]
```diff
@@ -138,7 +139,7 @@ public function &getItem($itemId = null)
 		}
 
 		$properties = $table->getProperties(1);
-		$value      = ArrayHelper::toObject($properties, 'JObject');
+		$value      = ArrayHelper::toObject($properties, 'CMSObject');
```
Comment (Quy) [inline administrator/components/com_menus/Model/MenuModel.php]:
```suggestion
		$value      = ArrayHelper::toObject($properties, CMSObject::class);
```

Comment (Quy) [reply administrator/components/com_menus/Model/MenuModel.php]: I don't know why, but it is done like this in other places.

Comment (Quy) [reply administrator/components/com_menus/Model/MenuModel.php]: See #21305

Comment (Quy) [reply administrator/components/com_menus/Model/MenuModel.php]: There are other instances that need to be updated.

Comment (brianteeman) [reply administrator/components/com_menus/Model/MenuModel.php]: In that case lets keep it as it is here and then change them all in their own PR

Comment (Quy) [reply administrator/components/com_menus/Model/MenuModel.php]: Here is the error message: `Class 'CMSObject' not found`

Comment (Quy) [reply administrator/components/com_menus/Model/MenuModel.php]: There are 6 instances to change. I can submit these changes in this PR if you want.

Comment (brianteeman) [reply administrator/components/com_menus/Model/MenuModel.php]: if you could that would be great - dont have much free time

File: administrator/components/com_languages/Model/LanguageModel.php [hunk 16]
```diff
@@ -124,7 +125,7 @@ public function getItem($langId = null)
 		}
 
 		$properties = $table->getProperties(1);
-		$value      = ArrayHelper::toObject($properties, 'JObject');
+		$value      = ArrayHelper::toObject($properties, 'CMSObject');
```
Comment (Quy) [inline administrator/components/com_languages/Model/LanguageModel.php]:
```suggestion
		$value      = ArrayHelper::toObject($properties, CMSObject::class);
```

File: administrator/components/com_modules/Model/ModuleModel.php [hunk 17]
```diff
@@ -718,7 +720,7 @@ public function getItem($pk = null)
 
 			// Convert to the \JObject before adding other data.
 			$properties        = $table->getProperties(1);
-			$this->_cache[$pk] = ArrayHelper::toObject($properties, 'JObject');
+			$this->_cache[$pk] = ArrayHelper::toObject($properties, 'CMSObject');
```
Comment (Quy) [inline administrator/components/com_modules/Model/ModuleModel.php]:
```suggestion
			$this->_cache[$pk] = ArrayHelper::toObject($properties, CMSObject::class);
```

File: administrator/components/com_plugins/Model/PluginModel.php [hunk 18]
```diff
@@ -177,7 +179,7 @@ public function getItem($pk = null)
 
 			// Convert to the \JObject before adding other data.
 			$properties = $table->getProperties(1);
-			$this->_cache[$pk] = ArrayHelper::toObject($properties, 'JObject');
+			$this->_cache[$pk] = ArrayHelper::toObject($properties, 'CMSObject');
```
Comment (Quy) [inline administrator/components/com_plugins/Model/PluginModel.php]:
```suggestion
			$this->_cache[$pk] = ArrayHelper::toObject($properties, CMSObject::class);
```

File: administrator/components/com_templates/Model/StyleModel.php [hunk 19]
```diff
@@ -365,7 +366,7 @@ public function getItem($pk = null)
 
 			// Convert to the \JObject before adding other data.
 			$properties        = $table->getProperties(1);
-			$this->_cache[$pk] = ArrayHelper::toObject($properties, 'JObject');
+			$this->_cache[$pk] = ArrayHelper::toObject($properties, 'CMSObject');
```
Comment (Quy) [inline administrator/components/com_templates/Model/StyleModel.php]:
```suggestion
			$this->_cache[$pk] = ArrayHelper::toObject($properties, CMSObject::class);
```

File: administrator/components/com_users/Model/GroupModel.php [hunk 20]
```diff
@@ -128,9 +130,9 @@ protected function loadFormData()
 	 * @since   1.6
 	 * @throws  \Exception if there is an error loading the form.
 	 */
-	protected function preprocessForm(\JForm $form, $data, $group = '')
+	protected function preprocessForm(Form $form, $data, $group = '')
 	{
-		$obj = is_array($data) ? ArrayHelper::toObject($data, 'JObject') : $data;
+		$obj = is_array($data) ? ArrayHelper::toObject($data, 'CMSObject') : $data;
```
Comment (Quy) [inline administrator/components/com_users/Model/GroupModel.php]:
```suggestion
		$obj = is_array($data) ? ArrayHelper::toObject($data, CMSObject::class) : $data;
```

File: administrator/components/com_menus/Model/MenuModel.php (hunk 15 above)
Comment (Quy) [reply administrator/components/com_menus/Model/MenuModel.php]: Done. 👍

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on bd1e5aff9c5e16470778b73971686a6236282d7f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24931">issues.joomla.org/tracker/joomla-cms/24931</a>.</sub>

**Comment by brianteeman** [comment]
@Quy sorry - I took advantage of the PR not being merged yet to include some more

File: libraries/src/MVC/Model/AdminModel.php [hunk 21]
```diff
@@ -1552,7 +1554,7 @@ public function initBatch()
 			$this->tableClassName = end($tc);
 
 			// Get UCM Type data
-			$this->contentType = new \JUcmType;
+			$this->contentType = new \UcmType;
```
Comment (infograf768) [inline libraries/src/MVC/Model/AdminModel.php]: \ toomuch ?

File: libraries/src/Table/CoreContent.php [hunk 22]
```diff
@@ -12,6 +12,7 @@
 
 use Joomla\CMS\Application\ApplicationHelper;
 use Joomla\CMS\Factory;
+use Joomla\CMS\Helper\HelperContent;
```
Comment (Quy) [inline libraries/src/Table/CoreContent.php]:
```suggestion
use Joomla\CMS\Helper\ContentHelper;
```

File: libraries/src/MVC/Model/AdminModel.php [hunk 23]
```diff
@@ -1552,7 +1554,7 @@ public function initBatch()
 			$this->tableClassName = end($tc);
 
 			// Get UCM Type data
-			$this->contentType = new \JUcmType;
+			$this->contentType = new UcmType;
```
Comment (Quy) [inline libraries/src/MVC/Model/AdminModel.php]:
```suggestion
			$this->contentType = new UCMType;
```

File: libraries/src/MVC/View/CategoryFeedView.php [hunk 24]
```diff
@@ -39,7 +41,7 @@ public function display($tpl = null)
 		$extension      = $app->input->getString('option');
 		$contentType = $extension . '.' . $this->viewName;
 
-		$ucmType = new \JUcmType;
+		$ucmType = new UcmType;
```
Comment (Quy) [inline libraries/src/MVC/View/CategoryFeedView.php]:
```suggestion
		$ucmType = new UCMType;
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on e7290ca3d38f150776fc9213beab673d87b43047<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24931">issues.joomla.org/tracker/joomla-cms/24931</a>.</sub>

**Comment by brianteeman** [comment]
@zero-24 @SniperSister Please could you look at rips for this PR

**Comment by zero-24** [comment]
Should be fixed by a new run: https://ci.joomla.org/joomla/joomla-cms/18526

**Comment by wilsonge** [comment]
Thanks!

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**